### DISCUSSION TITLE
How to prevent duplicates of accounts when allowing email logins as well as social logins?

### DISCUSSION THREAD

**Discussion by openscript** [ory/kratos#3064] [Q&A]
We encounter the problem, that users who sign up with e-mail address and password, can create another account with the same e-mail address, when they sign up via a social login provider.

1. User creates an account with example@gmail.com and a password.
2. A while later the same user signs up via Google OAuth. This account is linked to example@gmail.com as well.
3. Now two accounts with the same identifier is created.

How this can be prevented, or how the existing account can be linked to the Google account?

**Comment by aeneasr** [comment]
That's only possible to happen, when the JSON Schema or JsonNet is configured incorrectly. We do not allow duplicate values for fields that are considered identifiers! What do your configs look like?

**Reply by openscript** [reply]
My schema looks like this:

```
{
  "$id": "https://schemas.ory.sh/presets/kratos/quickstart/email-password/identity.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Customer",
  "type": "object",
  "properties": {
    "traits": {
      "type": "object",
      "properties": {
        "lang": {
          "type": "string",
          "title": "Language"
        },
        "tac": {
          "type": "string",
          "title": "Terms and Conditions"
        },
        "dp": {
          "type": "string",
          "title": "Data Protection"
        },
        "email": {
          "type": "string",
          "format": "email",
          "title": "Email",
          "minLength": 3,
          "ory.sh/kratos": {
            "credentials": {
              "password": {
                "identifier": true
              }
            },
            "verification": {
              "via": "email"
            },
            "recovery": {
              "via": "email"
            }
          }
        }
      },
      "required": [
        "email",
        "tac",
        "dp"
      ],
      "additionalProperties": false
    }
  }
}
```

Isn't it enough to make it unique by declaring the email as an identifier?
